server/src/main.rs

```diff
@@ -0,0 +1,248 @@
+    OnceCell::const_new();
+
+/// shorthand function to clone vehicle client
+pub fn get_vehicle_client() -> VehicleRpcClient<Channel> {
+    VEHICLE_CLIENT
+        .get()
+        .expect("Storage Client not initialized")
+        .clone()
+}
+
+/// shorthand function to clone vertiport client
+pub fn get_vertiport_client() -> VertiportRpcClient<Channel> {
+    VERTIPORT_CLIENT
+        .get()
+        .expect("Storage Client not initialized")
+        .clone()
+}
+
+/// shorthand function to clone flight plan client
+pub fn get_flight_plan_client() -> FlightPlanRpcClient<Channel> {
+    FLIGHT_PLAN_CLIENT
+        .get()
+        .expect("Storage Client not initialized")
+        .clone()
+}
+
+///Implementation of gRPC endpoints
+#[derive(Debug, Default, Copy, Clone)]
+pub struct SchedulerGrpcImpl {}
+
+#[tonic::async_trait]
+impl SchedulerRpc for SchedulerGrpcImpl {
+    ///finds the first possible flight for customer location, flight type and requested time.
+    /// Returns draft QueryFlightPlan which can be confirmed or cancelled.
+    async fn query_flight(
+        &self,
+        request: Request<QueryFlightRequest>,
+    ) -> Result<Response<QueryFlightResponse>, Status> {
+        let res = queries::query_flight(
+            request,
+            get_flight_plan_client(),
+            get_vehicle_client(),
+            get_vertiport_client(),
+        )
+        .await;
+        if res.is_err() {
+            error!("{}", res.as_ref().err().unwrap());
+        }
+        res
+    }
+
+    ///Confirms the draft flight plan by id.
+    async fn confirm_flight(
+        &self,
+        request: Request<Id>,
+    ) -> Result<Response<ConfirmFlightResponse>, Status> {
+        let res = queries::confirm_flight(request, get_flight_plan_client()).await;
+        if res.is_err() {
+            error!("{}", res.as_ref().err().unwrap());
+        }
+        res
+    }
+
+    ///Cancels the draft flight plan by id.
+    async fn cancel_flight(
+        &self,
+        request: Request<Id>,
+    ) -> Result<Response<CancelFlightResponse>, Status> {
+        let res = queries::cancel_flight(request, get_flight_plan_client()).await;
+        if res.is_err() {
+            error!("{}", res.as_ref().err().unwrap());
+        }
+        res
+    }
+
+    /// Returns ready:true when service is available
+    async fn is_ready(
+        &self,
+        _request: Request<ReadyRequest>,
+    ) -> Result<Response<ReadyResponse>, Status> {
+        let response = ReadyResponse { ready: true };
+        Ok(Response::new(response))
+    }
+}
+
+async fn init_router(mut vertiport_client: VertiportRpcClient<Channel>) {
+    let vertiports = vertiport_client
+        .vertiports(Request::new(SearchFilter {
+            search_field: "".to_string(),
+            search_value: "".to_string(),
+            page_number: 0,
+            results_per_page: 50,
+        }))
+        .await
+        .unwrap()
+        .into_inner()
```

Comment:
how's this unwrap handled if it's none or err?

Reply:
You're right - this was unhandled. Fixed now.